### DISCUSSION TITLE
.netcore and second part of tutorial

### DISCUSSION THREAD

**Discussion by PiotrKruk99** [dotnet/Silk.NET#695] [Q&A] [answered]
I can't compile second file of tutorial because of unint from C# v9 language specification. I use linux and .netcore that uses C# v8 specification. Is it intendent? Could be that more methods will not be working on .netcore as I will be learn more advance futures?

**Comment by HurricanKai** [comment]
All methods are .NET standard 2.0 compatible.
This is purely the tutorials making it easier to read code using newer features. On < C# 9.0 use IntPtr instead.

**Reply by PiotrKruk99** [reply]
Unfortunately it didn't work. Silk.NET.OpenGL.GL.BufferData takes uint as argument and don't accept IntPtr. I tried with uint but I've got 'silkNetTest: /home/void/glfw/src/window.c:755: glfwShowWindow: Assertion `window != NULL' failed.' error during compilation. On second computer I moved from 3.1 to .Net 6.0 and there it works fine.

**Reply by Perksey** [reply]
Yeah we’ve got no idea what’s causing this error, but it’s not related to your code, .NET Core, or C# 9 - I’m pretty sure it’s a bug in GLFW.

**Comment by PiotrKruk99** [comment, marked as answer]
Thanks. I will try with this :)

**Reply by HurricanKai** [reply]
If it works out for you, please mark the above as answer.
